### ISSUE TITLE
Status codes clarification

### ISSUE THREAD

**Issue by AnthonyGiretti** [grpc/grpc-dotnet#699]
Hi.

I understand that grpc requires HTTP 200 OK on POST.

But there is a thing I don't understand is this sentence within gRPC specs :(https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md)

"Implementations should expect broken deployments to send non-200 HTTP status codes in responses as well as a variety of non-GRPC content-types and to omit Status & Status-Message. Implementations must synthesize a Status & Status-Message to propagate to the application layer when this occurs."

What is your understanding of "broken deployments" ?

Another question in case of UNAUTHENTICATED response I get the proper gRPC status, but the Http status is 401. Are there any exceptions to the fact that gRPC always returns 200 OK?

**Comment by JamesNK** [comment]
1. An example of a broken deployment is an app with missing files. The web site can't startup so returns 500.
2. gRPC maps HTTP status codes to gRPC status codes in the client - https://github.com/grpc/grpc-dotnet/blob/89a95c33699fe26f5b6005479daca6ce2b98e835/src/Grpc.Net.Client/Internal/GrpcCall.cs#L279-L306

**Comment by AnthonyGiretti** [comment]
Thank you.
I will precise my question , because I don't understand.
Here is what I got some days ago during some tests.

1- A gRPC service works fine: I got gRPC status OK and HttpStatus OK, so no problem here
2- A gRPC services crashes with **unhandled error**: I got: gRPC status INTERNAL **but** HttpStatus 200 OK
3- gRPC service authentication fails:  I got: gRPC status UNAUTHENTICATED **and** HttpStatus 401

So my point is HttpStatus (and not gRPC status) seems unrealiable. Sometimes it's 200 even if there is a crash, sometimes it's consistent (UNAUTHENTICATED (grpc) / 401 (http))

I try to understand if I should not paying attention (ot not) to that status when I check HttpResponse from a grpc-dotnet server.
